optaplanner-core/src/test/java/org/optaplanner/core/impl/heuristic/selector/common/nearby/BetaDistributionNearbyRandomTest.java

```diff
@@ -40,14 +40,12 @@ public void nextIntUniform() {
         Random random = mock(Random.class);
         NearbyRandom nearbyRandom = new BetaDistributionNearbyRandom(1.0, 1.0);
 
-        when(random.nextDouble()).thenReturn(0.0);
-        assertEquals(0, nearbyRandom.nextInt(random, 500));
-        when(random.nextDouble()).thenReturn(1.0 / 500.0);
-        assertEquals(1, nearbyRandom.nextInt(random, 500));
-        when(random.nextDouble()).thenReturn(2.0 / 500.0);
-        assertEquals(2, nearbyRandom.nextInt(random, 500));
-        when(random.nextDouble()).thenReturn(3.0 / 500.0);
-        assertEquals(3, nearbyRandom.nextInt(random, 500));
+        double roundingConstant = 0.000001; // normalizes calculation against over-rounding error
+        for (int i = 0; i < 500; i++) {
```

Comment:
DRY (Do not repeat yourself), extract methods, etc is a must do in production code, but I don't think it's a must do in unit test code, at least not to calculate the expected result of an assert.